source/fundamentals/data-formats/documents.txt

```diff
@@ -284,6 +289,94 @@ For more information on retrieving and manipulating MongoDB data, see our
 
+JsonObject
+-------------
+
+The ``JsonObject`` class acts as a wrapper for JSON strings.
+``JsonObject`` is a more performant alternative to both ``Document``
+and ``BsonDocument`` when working with JSON strings because it does not create an intermediate
+``Map`` object.
+
+By default, ``JsonObject`` stores :manual:`Extended JSON </reference/mongodb-extended-json/>`.
+You can customize the format of JSON in ``JsonObject`` by specifying a
+:java-docs:`JsonObjectCodec</apidocs/bson/org/bson/codecs/JsonObjectCodec.html>`
+and passing it a
+:java-docs:`JsonWriterSettings</apidocs/bson/org/bson/json/JsonWriterSettings.html>`
+object. For more information on JSON formats, see
+our :doc:`Extended JSON guide </fundamentals/data-formats/document-data-format-extended-json/>`.
+For more information on specifying codecs, see our 
+:doc:`Codecs guide </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate a sample ``JsonObject`` 
+instance wrapping an Extended JSON string containing different types of key value pairs:
+
+.. code-block:: java
+
+   String ejsonStr =  "{"
+                     +" \"_id\": {"
+                     +"  \"$oid\": \"6035210f35bd203721c3eab8\""
+                     +" },"
+                     +" \"name\": \"Gabriel Garc\\u00eda M\\u00e1rquez\","
+                     +" \"dateOfDeath\": {"
+                     +"  \"$date\": \"2014-04-17T04:00:00Z\""
+                     +" },"
+                     +" \"novels\": ["
+                     +"  {"
+                     +"   \"title\": \"One Hundred Years of Solitude\","
+                     +"   \"yearPublished\": 1967"
+                     +"  },"
+                     +"  {"
+                     +"   \"title\": \"Chronicle of a Death Foretold\","
+                     +"   \"yearPublished\": 1981"
+                     +"  },"
+                     +"  {"
+                     +"   \"title\": \"Love in the Time of Cholera\","
+                     +"   \"yearPublished\": 1985"
+                     +"  }"
+                     +" ]"
+                     +"}";
+   JsonObject author = new JsonObject(ejsonStr);
+
+To insert this document into a collection, instantiate a collection using
+:java-docs:`getCollection()</apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`,
+specifying the ``JsonObject`` class. Then, call the
+:doc:`insertOne </usage-examples/insertOne>` operation as follows:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+   MongoCollection<JsonObject> collection = database.getCollection("authors", JsonObject.class);
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample JSON data from the 
+collection using the following code:
```

Comment:
Nitpick: I would recommend mentioning whether you need to use a JsonObject for the query as I'm not sure it was mentioned earlier and could be confusing.

Reply:
Thanks for the feedback. This language is directly copied from the other three sections of the page. For example here is the version for the [BasicDBObject section](https://github.com/biniona-mongodb/docs-java/blob/master/source/fundamentals/data-formats/documents.txt#L331). Would you like the four sections updated to resemble:

```suggestion
Once you perform a successful insert, you can retrieve the sample JSON data from the 
collection using using a `JsonObject` query:
```

Reply:
Sorry, after investigating the driver codebase, it looks like this only needs to be an object that is a subclass of `Bson` which means you can use a `JsonObject`, `BsonDocument`, `Document`, `Filter` class helper method, etc. I would probably change this sentence and all similar instances to indicate that you're specifically showing how to query using that specific class. E.g.
While you can use any class that extends `Bson` to specify your query, here is how to query your data using a `JsonObject`:

Reply:
Updated the language around using a `JsonObject` query here https://github.com/mongodb/docs-java/pull/57/commits/1b26a7ec186aab78339cbe21acf055165db0aac9. I only did it for `JsonObject` as the other sections use the `Filters` class to query for results. I missed this at the time of writing, but I think it is nice that `JsonObject` uses JSON for its query as it is for people interested in working with JSON strings.